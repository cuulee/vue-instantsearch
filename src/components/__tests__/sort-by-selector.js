import Vue from 'vue';
import SortBySelector from '../SortBySelector.vue';

test('renders proper HTML', () => {
  const searchStore = {};
  const Component = Vue.extend(SortBySelector);
  const vm = new Component({
    propsData: {
      searchStore,
      indices: [
        { name: 'index1', label: 'label1' },
        { name: 'index2', label: 'label2' },
      ],
    },
  });
  vm.$mount();

  expect(vm.$el.outerHTML).toMatchSnapshot();
});

test('should use current index name in search store if it is a valid option', () => {
  const searchStore = {
    indexName: 'index2',
  };
  const Component = Vue.extend(SortBySelector);
  const vm = new Component({
    propsData: {
      searchStore,
      indices: [
        { name: 'index1', label: 'label1' },
        { name: 'index2', label: 'label2' },
      ],
    },
  });

  expect(vm.searchStore.indexName).toEqual('index2');

  vm.$mount();

  expect(vm.$el.getElementsByTagName('option')[0].selected).toEqual(false);
  expect(vm.$el.getElementsByTagName('option')[1].selected).toEqual(true);
});

test('should use first index if current index name is not a valid option', () => {
  const searchStore = {
    indexName: 'index3',
  };
  const Component = Vue.extend(SortBySelector);
  const vm = new Component({
    propsData: {
      searchStore,
      indices: [
        { name: 'index1', label: 'label1' },
        { name: 'index2', label: 'label2' },
      ],
    },
  });

  expect(vm.searchStore.indexName).toEqual('index1');

  vm.$mount();

  expect(vm.$el.getElementsByTagName('option')[0].selected).toEqual(true);
});
