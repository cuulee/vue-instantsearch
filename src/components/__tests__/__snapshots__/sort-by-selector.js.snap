// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders proper HTML 1`] = `

<select class="ais-sort-by-selector">
  <option value="index1">
    label1
  </option>
  <option value="index2">
    label2
  </option>
</select>

`;
